packages/pigeon/lib/generator_tools.dart

```diff
@@ -801,6 +809,24 @@ String toUpperCamelCase(String text) {
   }).join();
 }
 
+/// Converts strings to Lower Camel Case.
+String toLowerCamelCase(String text) {
+  final RegExp separatorPattern = RegExp(r'[ _-]');
+  bool firstWord = true;
+  return text.split(separatorPattern).map((String word) {
+    if (word.isEmpty) {
+      return '';
+    }
+    if (firstWord) {
+      firstWord = false;
+      return word.substring(0, 1).toLowerCase() + word.substring(1);
+    }
+    return word.isEmpty
```

Comment:
The `isEmpty` codepath is dead code now, due to the new early return up top. So this can just `return word.substring[...]`.